pkg/sources/s3/s3.go

```diff
@@ -336,15 +335,27 @@ func (s *Source) pageChunker(ctx context.Context, client *s3.S3, chunksChan chan
 			reader.Stop()
 
 			chunk := *chunkSkel
-			chunkData, err := io.ReadAll(reader)
-			if err != nil {
-				s.log.Error(err, "Could not read file data.")
+			chunkReader := sources.NewChunkReader()
+			dataCh, errCh := chunkReader(ctx, reader)
+			for data := range dataCh {
+				chunk.Data = data
+				select {
+				case <-ctx.Done(): // priority to context cancellation
+					return ctx.Err()
+				default:
+					select {
+					case <-ctx.Done():
+						return ctx.Err()
+					case chunksChan <- &chunk:
+					}
+				}
```

Comment:
Cool -- would you also mind adding this to [common.CancellableWrite](https://github.com/trufflesecurity/trufflehog/blob/v3.46.3/pkg/common/context.go#L14-L24)?